src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/DefaultVSCredentialServiceProvider.cs

```diff
@@ -20,41 +22,38 @@ public class DefaultVSCredentialServiceProvider : ICredentialServiceProvider
     {
 
         private readonly Lazy<INuGetUILogger> _outputConsoleLogger;
-        private readonly IServiceProvider _serviceProvider;
+        private readonly IAsyncServiceProvider _asyncServiceProvider;
 
         [ImportingConstructor]
+        internal DefaultVSCredentialServiceProvider(Lazy<INuGetUILogger> outputConsoleLogger)
+            : this(AsyncServiceProvider.GlobalProvider, outputConsoleLogger)
+        { }
+
         internal DefaultVSCredentialServiceProvider(
-           [Import(typeof(SVsServiceProvider))]
-            IServiceProvider serviceProvider,
-           Lazy<INuGetUILogger> outputConsoleLogger
+            IAsyncServiceProvider asyncServiceProvider,
+            Lazy<INuGetUILogger> outputConsoleLogger
             )
         {
-            if (serviceProvider == null)
-            {
-                throw new ArgumentNullException(nameof(serviceProvider));
-            }
-            if (outputConsoleLogger == null)
-            {
-                throw new ArgumentNullException(nameof(outputConsoleLogger));
-            }
-            _serviceProvider = serviceProvider;
-            _outputConsoleLogger = outputConsoleLogger;
+            _asyncServiceProvider = asyncServiceProvider ?? throw new ArgumentNullException(nameof(asyncServiceProvider));
+            _outputConsoleLogger = outputConsoleLogger ?? throw new ArgumentNullException(nameof(outputConsoleLogger));
         }
 
         [SuppressMessage("Microsoft.VisualStudio.Threading.Analyzers", "VSTHRD010", Justification = "NuGet/Home#4833 Baseline")]
-        public NuGet.Configuration.ICredentialService GetCredentialService()
+        public async Task<NuGet.Configuration.ICredentialService> GetCredentialServiceAsync()
         {
             ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
does it still need to be on UI thread?

Reply:
it should be ok on worker thread, I can remove this